src/Controller/WalletController.php

```diff
@@ -106,9 +107,10 @@ public function useCreditAction(Request $request): Response
         $orderRepository = $this->container->get('sylius.repository.order');
         $order = $orderRepository->findCartByTokenValue($request->get('token'));
         $amount = $walletService->useWallet($order, $request->get('amount'));
-        return new JsonResponse([
-            'amount' => $amount / 100
-        ]);
+
+        $response = $walletService->getCart($request->get('token') , $amount);
```

Comment:
An extra query will be run if user is using the web version, so I think you should return the wallet in shop-api